application/src/test/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceTest.java

```diff
@@ -24,73 +22,68 @@
+        .thenReturn(postcodeMappingOptList);
+    List<PostcodeMapping> findByPostCodeIn =
+        postcodeMappingService.getByPostCodesAndName(postCodes, serviceName);
+    assertTrue(findByPostCodeIn.size() > 0);
+  }
 
-    @Test
-    public void testGetByCcgName() throws InvalidPostcodeException
-    {
-        when(postcodeMappingRepository.findByName("Nhs Halton CCG")).thenReturn(postcodeMappingOptList);
-        List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getByName("Nhs Halton CCG");
-        assertTrue(findByPostCodeIn.size() > 0);
-    }
+  @Test
+  public void testGetByPostCodes() throws InvalidPostcodeException {
+    when(postcodeMappingRepository.findByPostCodeIn(postCodes)).thenReturn(postcodeMappingOptList);
+    List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getByPostCodes(postCodes);
+    assertTrue(findByPostCodeIn.size() > 0);
+  }
 
-    @Test
-    public void testGetAll() throws InvalidPostcodeException
-    {
-        when(postcodeMappingRepository.findAll()).thenReturn(postcodeMappingList);
-        List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getAll();
-        assertTrue(findByPostCodeIn.size() > 0);
-    }
+  @Test
+  public void testGetByCcgName() throws InvalidPostcodeException {
+    when(postcodeMappingRepository.findByName(serviceName)).thenReturn(postcodeMappingOptList);
+    List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getByName(serviceName);
+    assertTrue(findByPostCodeIn.size() > 0);
+  }
 
+  @Test
+  public void testGetAll() throws InvalidPostcodeException {
+    when(postcodeMappingRepository.findAll()).thenReturn(postcodeMappingList);
+    List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getAll();
+    assertTrue(findByPostCodeIn.size() > 0);
+  }
 
-    @Test
-    public void testEastingAndNorthing() throws InvalidPostcodeException
-    {
-        when(postcodeMappingRepository.findByName("Nhs Halton CCG")).thenReturn(postcodeMappingOptList);
-        List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getByName("Nhs Halton CCG");
-        PostcodeMapping postCodeMapping = findByPostCodeIn.stream().findAny().get();
-        int easting = postCodeMapping.getEasting();
-        int northing = postCodeMapping.getNorthing();
-        assertEquals(12345, easting);
-        assertEquals(360, northing);
-    }
+  @Test
+  public void testEastingAndNorthing() throws InvalidPostcodeException {
+    when(postcodeMappingRepository.findByName(serviceName)).thenReturn(postcodeMappingOptList);
+    List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getByName(serviceName);
+    PostcodeMapping postCodeMapping = findByPostCodeIn.stream().findAny().get();
+    int easting = postCodeMapping.getEasting();
+    int northing = postCodeMapping.getNorthing();
+    assertEquals(12345, easting);
+    assertEquals(360, northing);
+  }
 }
```

Comment:
Need to test an error scenario

Reply:
The service layer is mocked, so all the negative scenarios is included in Util test class (PostcodeUtils)